src/lib/gadgets/gadgets.ts

```diff
@@ -106,6 +106,49 @@ const Gadgets = {
   xor(a: Field, b: Field, length: number) {
     return xor(a, b, length);
   },
+
+  /**
+   * Bitwise NOT gate on {@link Field} elements. Similar to the [bitwise
+   * NOT `~` operator in JavaScript](https://developer.mozilla.org/en-US/docs/
+   * Web/JavaScript/Reference/Operators/Bitwise_NOT).
+   *
+   * **Note:** The NOT gate only operates over the amount
+   * of bits specified by the 'length' paramenter.
+   *
+   * A NOT gate works by returning `1` in each bit position if the
+   * corresponding bit of the operand is `0`, and returning `0` if the
+   * corresponding bit of the operand is `1`.
+   *
+   * The `length` parameter lets you define how many bits to NOT.
+   *
+   * **Note:** Specifying a larger `length` parameter adds additional constraints.
+   *
+   *
+   * NOT is implemented in two different ways. If the 'checked' parameter is set to 'true'
+   * the {@link Gadgets.xor} gadget is reusesd the with a second argument to be an
+   * all one bitmask the same length. This approach needs as many rows as an XOR would need
+   * for a single negation. If the 'checked' parameter is set to 'false' NOT is
+   * implementad as a subtraction of the input from the all one bitmask. This
+   * implementation is returned by default if no 'checked' parameter is provided.
+   *
+   *You can find more details about the implementation in the [Mina book](https://o1-labs.github.io/proof-systems/specs/kimchi.html?highlight=gates#not)
+   *
+   * @example
+   * ```ts
+   * let a = Field(0b0101);
+   * let b = not(a,4,false);    // not-ing 4 bits
+   *
+   * b.assertEquals(0b1010);
+   * ```
+   *
+   * @param a - The value to apply NOT to.
+   * @param length - The number of bits to be considered for the NOT operation.
+   * @param checked - Optional Boolean to determine if the checked or unchecked not implementation is used.
```

Comment:
We should clarify what this means for the developer and how to utilize the two variants